### PR TITLE
[AST] Break dependency cycles of swiftAST against swiftSema and swiftClangImporter

### PR DESCRIPTION
**Pull request by akyrtzi**

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci smoke test

**Comment by jrose-apple** [comment]
Huh, I would have gone the other way with the Sema one, sinking `isExtensionApplied` into AST. I don't like splitting up lookupVisibleDecls from lookup-by-name.

**Comment by akyrtzi** [comment]
> Huh, I would have gone the other way with the Sema one, sinking `isExtensionApplied` into AST. I don't like splitting up lookupVisibleDecls from lookup-by-name.

Could you clarify how I could do that ? That function uses the typechecker, it needs to be in Sema AFAICT.

**Comment by jrose-apple** [comment]
Oh. That's…unfortunate. Okay, let me take another look.

File: lib/AST/NameLookup.cpp [hunk 1]
```diff
@@ -2690,3 +2694,197 @@ swift::getDirectlyInheritedNominalTypeDecls(
 
   return result;
 }
+
+namespace swift {
+namespace namelookup {
```
Comment (jrose-apple) [inline lib/AST/NameLookup.cpp]: Nitpick: You don't need these in this file; just qualify things with `namelookup::`. Or even adding a `using namespace swift::namelookup` or `using namelookup::FindLocalVal` at the top.

**Review by jrose-apple — APPROVED** [review]
Okay, I guess this is indeed the least bad option.

File: lib/AST/NameLookup.cpp (hunk 1 above)
Comment (akyrtzi) [reply lib/AST/NameLookup.cpp]: Good point, done.

**Comment by akyrtzi** [comment]
@swift-ci smoke test and merge

**Pull request merged**